src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateGetHashCodeMethod.cs

```diff
@@ -179,11 +179,19 @@ public static IMethodSymbol GetBaseGetHashCodeMethod(INamedTypeSymbol containing
                 return thisSymbol;
             }
 
-            var memberType = member.GetSymbolType();
-            var primitiveValue = IsPrimitiveValueType(memberType) && memberType.SpecialType != SpecialType.System_String;
-            var isTupleType = memberType?.IsTupleType == true;
-            if (primitiveValue || isTupleType)
+            if (member.GetSymbolType().IsValueType)
             {
+                /* No reason to use the bulkier EqualityComparer<>.Default.GetHashCode. No null check is necessary, and
+                calling GetHashCode directly will not cause boxing. (The decision to box is at jit time. If boxing
+                happens, there will have been no way to avoid it anyway.)
```

Comment:
I'm not entirely sure I'm following this comment. I'd assume the JIT won't actually be doing a box and would elide the null check since it'll be doing a custom JIT for value types anyways (there's no System.__Canon version it can use). Are we sure this is actually faster? This feels like something we'd tell the runtime to do instead...?

Reply:
You're right. I don't know the right term to use instead of 'boxing.'

> Are we sure this is actually faster? This feels like something we'd tell the runtime to do instead...?

Not faster, but equivalent and nicer-looking.

.NET Core x64: [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABABgAJiBGFAbgFgAoJ4gJnIGEmBvJ8/8gAcoASwBu2DDHKxsAEwgA7ADYBPcgHURGABYB5MTCii50gO7b9h4yNMNGAoaIlSZMeUrWbLEAK4YDIxNzH39Amzs+ASj+YgBmchFFDHIAcRgMAAk8HQ4IUy1dcOCACgBKGPJeB0cBYgB2cgsi62CAOnSsnLzTcvtHAF9KyvjE5PIAUQBHX2xlbVU8/EFsWChCqyDbGHLK6tq6xunZ+YxFiGXVowAeDeLtgD42gBEYADNsX2UMDozs3Fy+R2zU2ERgZX6AiGNX4IwSSRSnX+gIKoQCrW2uxhVUqjgaTTR91Mvy6AJ6OwhlWhjjhYxSxzmCyWKzWGz86K2vQq2P2B0oRxmjLOzKuUFuhIxpierw+Xx+SO6QJKIPZRPBkP40OhTFwGCgvjAKTukpgPFpCPIAAUoBBBEYzlVyABzDK0cja7GjCAmulpP6KrnkAC8DytNrtUDO9g9uv1hu8ulVJrNnvh42ttvt6m4ztd7qY0KAA===)

.NET Framework x64: [sharplab](https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAGABJgRigG4BYAKAswCZcBhCgbwt1dwAcAnASwDdE4AU1ydBiACYB7AHYAbAJ64A6tzgALAPK9BnHuOEB3VZu27u+suTYce/ISLFS5ileskBXOFp17Dxj16mvpbWLGyYAMy43NJwuADignAAEihqdJL6riY+5oIAFACUYazMVtbhAOy4RureZvoAdIkpaRn6RSFsAL4luH2R0bG4AKIAju6IsqryGajsiKKc2fW+RX1lFVWjE1MzcwtLADwrQXkAfI0AIoIAZojusnDNSanI6ZkFtTkNgoVdrF65VYAyiMTiLTeHyy/k8qzy62BuE2W3w1W+AXhTUhbU+nT6QNCSMG4J2k2mcFmknmix02UxZw6xSRKK2mGq43J+2phx0J1hgVy+kuN3uj2eOPe7S+AqxfwBuCBQIoyDgnHcAGM4qchYImKChnEAAqcSTsHSU5G4ADmSWIiooBskjOEpMl0IKhVwAF5zrgTWaLfJLMryKr1VrlLKXfriWDhgHzZxLYwbXaHeQgUA===) (EqualityComparer.Default actually gens much worse code than calling GetHashCode directly)

Reply:
System.HashCode is used to implement GetHashCode on .NET Core, so if this code is running, we're on .NET Framework or .NET Standard etc. The poorer performance on .NET Framework seems like it would be another reason to avoid using EqualityComparer<>.Default on value types.